app/frontend/src/features/messages/groupchats/GroupChatView.tsx

```diff
@@ -120,12 +151,47 @@ export default function GroupChatView() {
 
   const handleBack = () => history.goBack();
 
+  const scrollRef = useRef<HTMLElement>(null);
+  const prevScrollHeight = useRef<number | undefined>(undefined);
+  const prevTopMessageId = useRef<number | null>(null);
+
+  const { ref: loadMoreRef } = useOnVisibleEffect(null, () => {
+    prevScrollHeight.current = scrollRef.current?.scrollHeight;
+    if (messagesRes) {
+      prevTopMessageId.current =
+        messagesRes.pages[messagesRes.pages.length - 1].lastMessageId;
+    }
+    fetchNextPage();
+  });
+
+  useEffect(() => {
+    if (isFetchingNextPage) return;
+    document
+      .getElementById(`message-${prevTopMessageId.current}`)
+      ?.scrollIntoView();
+    //make up for circular progress offset
+    scrollRef.current?.scrollBy(0, -60);
+    /*if (isFetchingNextPage || !scrollRef.current) return;
+    const newScrollHeight = scrollRef.current.scrollHeight;
+    if (newScrollHeight === prevScrollHeight.current) return;
+    scrollRef.current.scrollTo(
+      0,
+      newScrollHeight - (prevScrollHeight.current ?? newScrollHeight)
+    );
+    prevScrollHeight.current = newScrollHeight;*/
+  }, [isFetchingNextPage]);
```

Comment:
@darrenvong what do you think of this? It's difficult doing infinite load "upwards"! The current way is less janky than the commented-out one, but still not amazing. And I don't really like using id like that.

Reply:
What happens with this when the next page loads? Does it scroll all the way to the oldest message on that page? Can't really think of a cleaner way to do this and have it scroll in smoothly, as I doubt we can do better than the browser with `scrollIntoView`. Would the `smooth` option help make this less janky too?